File: app/src/main/java/com/wxm/camerajob/ui/Utility/Setting/TFSettingCheckVersion.java
package com.wxm.camerajob.ui.Utility.Setting;


import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.wxm.camerajob.R;
import com.wxm.camerajob.utility.utility.ContextUtil;
import com.wxm.camerajob.ui.Base.TFSettingBase;

import java.util.Locale;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * 检查版本设置页面
 * Created by 123 on 2016/10/10.
 */
public class TFSettingCheckVersion extends TFSettingBase {
    @BindView(R.id.tv_show)
    TextView mTVShow;

    @Override
    protected View inflaterView(LayoutInflater inflater, ViewGroup container, Bundle bundle) {
        LOG_TAG = "TFSettingCheckVersion";
        View rootView = inflater.inflate(R.layout.frg_setting_version, container, false);
        ButterKnife.bind(this, rootView);
        return rootView;
    }

    @Override
    protected void initUiComponent(View view) {
        String s = String.format(Locale.CHINA
                , "当前版本号 : %d\n当前版本名 : %s"
                , ContextUtil.getVerCode(getContext())
                , ContextUtil.getVerName(getContext()));
        mTVShow.setText(s);
    }

    @Override
    protected void initUiInfo() {
    }

    @Override
    public void updateSetting() {
        if (mBSettingDirty) {
            mBSettingDirty = false;
        }
    }
}

File: app/src/main/java/com/wxm/camerajob/ui/Utility/Setting/TFSettingCheckVersion.kt
package com.wxm.camerajob.ui.Utility.Setting

import android.os.Bundle
import android.widget.TextView

import com.wxm.camerajob.R
import com.wxm.camerajob.utility.ContextUtil

import java.util.Locale

import kotterknife.bindView

/**
 * UI for check version
 * Created by WangXM on 2016/10/10.
 */
class TFSettingCheckVersion : TFSettingBase() {

    private val mTVShow: TextView by bindView(R.id.tv_show)

    override fun isUseEventBus(): Boolean = false
    override fun getLayoutID(): Int = R.layout.frg_setting_version

    override fun initUI(savedInstanceState: Bundle?) {
          mTVShow.text =  String.format(Locale.CHINA,
                  "当前版本号 : %d\n当前版本名 : %s",
                  ContextUtil.getVerCode(context), ContextUtil.getVerName(context))
    }

    override fun updateSetting() {
        if (isSettingDirty) {
            isSettingDirty = false
        }
    }
}
